data/data/azure/vnet/variables.tf

```diff
@@ -55,3 +55,32 @@ variable "private_dns_zone_id" {
   type        = string
   description = "This is to create explicit dependency on private zone to exist before VMs are created in the vnet. https://github.com/MicrosoftDocs/azure-docs/issues/13728"
 }
+
+variable "preexisting_network" {
+  type    = bool
+  default = false
+}
+
+variable "virtual_network" {
+  type = string
+}
+
+variable "network_resource_group" {
+  type = string
+}
+
+variable "master_subnet" {
+  type = string
+}
+
+variable "worker_subnet" {
+  type = string
+}
+
+variable "master_subnet_id" {
```

Comment:
as the VNet module is the one that creates the subnet, or sources existing subnet, it cannot possibly take the id as input.

Reply:
I've fixed this.